pkg/activator/handler/handler.go

```diff
@@ -34,19 +35,44 @@ type ActivationHandler struct {
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
 	Reporter  activator.StatsReporter
+	Throttler *activator.Throttler
+}
+
+type ErrorResponse struct {
+	Error string `json:"error"`
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
 
 	start := time.Now()
+	revID := activator.RevisionID{namespace, reconciler.GetServingK8SServiceNameForObj(name)}
+	a.Logger.Debug(revID)
+	a.Throttler.AddIfAbsent(revID)
+	// ActiveEndpoint() will block until the first endpoint is available
+	ar := a.Activator.ActiveEndpoint(namespace, name)
+	err := a.Throttler.Try(revID, func() {
+		a.activate(w, r, namespace, name, start, ar)
+	})
+	if err != nil {
+		error := ErrorResponse{Error: err.Error()}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusServiceUnavailable)
+		b, err := json.Marshal(error)
+		if err != nil {
+			a.Logger.Error(err)
+		}
+		w.Write(b)
```

Comment:
We use plaintext responses everywhere that I'm aware of. Let's not introduce JSON in this PR.

Also: Let's return a generic message here, not the error returned by the guts of the activator.